tests/lib/rules/display-name.js

```diff
@@ -894,5 +908,21 @@ ruleTester.run('display-name', rule, {
     errors: [{
       message: 'Component definition is missing display name'
     }]
+  }, {
+    code: `
+      const x = {
+        title: "URL",
+        dataIndex: "url",
+        key: "url",
+        render: url => (
```

Comment:
I don't think that under any conditions this function should be considered a component. Maybe if the name were uppercase, since that is a pattern I have seen in some libraries, something like:

```jsx
<MyComponent.Button />
```

But not this way. What do you think @ljharb ?

Reply:
I thought of the same thing after your last comment on the issue :D  

I am a little new to this codebase. could you please point me to the file/logic where we're classifying the code as a component?

Reply:
Yes, I agree. This shouldn't be a component.